src/main/kotlin/note/application/NoteDeleter.kt

```diff
@@ -15,7 +15,7 @@ class NoteDeleter(
      * @return Unit if the note was deleted or throws an exception if it does not exist.
      * @throws NonExistentNoteException if the note does not exist or is not saved on the repository.
      */
-    fun delete(identifier: NoteIdentifier) =
+    fun delete(identifier: Identifier) =
```

Comment:
This was addressed on the #42 change. 
I know that, since we didn't have any project rules, doing pull requests was a party which everyone could join.

As @oriolagobat stated, we must create this project rules or we will go nuts. I started a discussion on #43

Reply:
Yes, I agree. The need for the refactor to be done in order to get Note instances on the use case test suite ended with me doing the refactor on this pull request instead of creating a separate one. I'll remove those changes from this pull request and follow the project rules next time.